src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -163,10 +163,10 @@ public JsonConverter GetConverter(Type typeToConvert)
 
             if (converter != null)
             {
-                Type converterTypeToConvert = converter.TypeToConvert;
+                Type? converterTypeToConvert = converter.TypeToConvert;
```

Comment:
OK, reasoning through this one, there is no way for `converter.TypeToConvert` to be null here. We can remove this null check and use `!` here.

We should go even farther and use `null!` on JsonConverter and not make `TypeToConvert` nullable.

Reply:
ok